pkg/storageincentives/staking/contract.go

```diff
@@ -83,6 +86,134 @@ func New(
 	}
 }
 
+func (c *contract) DepositStake(ctx context.Context, stakedAmount *big.Int) (common.Hash, error) {
+	prevStakedAmount, err := c.GetPotentialStake(ctx)
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	if len(prevStakedAmount.Bits()) == 0 {
+		if stakedAmount.Cmp(MinimumStakeAmount) == -1 {
```

Comment:
why == -1 and not <= -1 ?

Reply:
-1 means that the staked amount if less than the minimum,
0 would mean equal
1 would mean staked amount is greater than the minimum

Reply:
the cmp strictly returns the values above

Reply:
my mistake